de.dlr.sc.virsat.model.extension.mechanical.ui/src/de/dlr/sc/virsat/model/extension/mechanical/ui/wizards/ACatiaImportExportPage.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.mechanical.ui.wizards;
+
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.FileDialog;
+
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.model.extension.ps.model.AssemblyTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementOccurence;
+import de.dlr.sc.virsat.project.ui.contentProvider.VirSatFilteredWrappedTreeContentProvider;
+import de.dlr.sc.virsat.uiengine.ui.wizard.AImportExportPage;
+
+/**
+ * A basic import / export page to select a tree element and a file destination for an import / export JSON file
+ *
+ */
+public abstract class ACatiaImportExportPage extends AImportExportPage {
+
+	private static final String DIALOG_TEXT = "File name";
+	private static final String[] DIALOG_EXTENSIONS = { "*.json" };
+	
+	/**
+	 * Default constructor override
+	 * @param pageName the page name
+	 */
+	protected ACatiaImportExportPage(String pageName) {
+		super(pageName);
+	}
+	
+	/**
+	 * Create a tree viewer with filters to show only relevant tree elements for CATIA import /export
+	 */
+	protected void createTreeViewer() {
+		TreeViewer treeViewer = createTreeUI();
+		VirSatFilteredWrappedTreeContentProvider filteredCp = (VirSatFilteredWrappedTreeContentProvider) treeViewer.getContentProvider();
+		filteredCp.addStructuralElementIdFilter(ConfigurationTree.FULL_QUALIFIED_STRUCTURAL_ELEMENT_NAME);
+		filteredCp.addStructuralElementIdFilter(ElementConfiguration.FULL_QUALIFIED_STRUCTURAL_ELEMENT_NAME);
+		filteredCp.addStructuralElementIdFilter(AssemblyTree.FULL_QUALIFIED_STRUCTURAL_ELEMENT_NAME);
+		filteredCp.addStructuralElementIdFilter(ElementOccurence.FULL_QUALIFIED_STRUCTURAL_ELEMENT_NAME);
+	}
+	
+	@Override
+	protected String openDialog() {
+		FileDialog dialog = new FileDialog(getContainer().getShell(), SWT.SAVE | SWT.SHEET);

```

Comment:
Can we put this into the general importexport page like this?
I seem to recall that there was a separate flag SWT.OPEN for dialogs that do opening / importing of stuff.
Maybe check out how it looks like, if the SWT.SAVE style is good for both then we can leave it.